### PR TITLE
[Graph/OperatorTest] Add weightedSum operator

### PR DESCRIPTION
**Pull request by jfix71**
Another operator used by seq2seq. I implemented it as a series of splats/multiplies/adds.

### CODE REVIEW

File: lib/Graph/Graph.cpp [hunk 1]
```diff
@@ -1131,6 +1131,36 @@ RescaleQuantizedNode *Function::createRescaleQuantized(llvm::StringRef name,
...
 }
 
+Node *Function::createWeightedSum(llvm::StringRef name,
+                                  llvm::ArrayRef<NodeValue> data,
+                                  llvm::ArrayRef<float> weights) {
+  assert(data.size() == weights.size() &&
+         "Must have same number of data and weights.");
+  assert(data.size() > 0 && "No inputs provided.");
+
+  auto *outTy = data[0].getType();
```
Comment (rdzhabarov) [inline lib/Graph/Graph.cpp]: do we need to assert all data[i] have the same type?

File: lib/Graph/Graph.cpp [hunk 2]
```diff
@@ -1131,6 +1131,36 @@ RescaleQuantizedNode *Function::createRescaleQuantized(llvm::StringRef name,
...
+
+  // Create a zero splat to bootstrap the adding chain.
+  Node *currAdd = createSplat(name.str() + ".splat", outTy, 0.);
+
+  for (size_t i = 0; i < data.size(); i++) {
```
Comment (rdzhabarov) [inline lib/Graph/Graph.cpp]: super nit: llvm coding standards prefer (i = 0, e = data.size(); i < e; i++)

File: lib/Graph/Graph.cpp (hunk 1 above)
Comment (jfix71) [reply lib/Graph/Graph.cpp]: I expect that to hit an assert elsewhere (e.g. in `createMul()`) if it wasn't the case, but it can't hurt to add here too!

**Comment by jfix71** [comment]
I changed this a bit -- the weights are not floats provided at compile time, they are calculated during execution and so instead need to be Tensors of a single element representing each weight. So instead of a splat of the float weight, I broadcast the Node weight to the correct shape.

**Pull request merged**